bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypedescProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.jvm;
+
+import org.ballerinalang.jvm.types.BTypes;
+import org.ballerinalang.jvm.values.RefValue;
+import org.ballerinalang.jvm.values.TypedescValue;
+
+/**
+ * Runtime impl to create typedesc values out of Ballerina values.
+ *
+ * @since 0.995.0
+ */
+public class TypedescProvider {
+    public static TypedescValue typeof(Object value) {
+        if (value instanceof RefValue) {
```

Comment:
Can use the `TypeChecker.getType(...)` method to get the type.

Reply:
We can also then move this method to the TypeChecker class it self, since it will be a one-liner :)

Reply:
With that we can codegen to invoke TypeChecker.getTypedesc(), skipping this class altogether right?

Reply:
yes